File: src/main/kotlin/com/stannis/dataModel/statementTypes/ArraySubscript.kt
package com.stannis.dataModel.statementTypes

import com.stannis.dataModel.Arguments
import com.stannis.dataModel.NameInterface
import com.stannis.dataModel.Statement

data class ArraySubscript(
    override val `$type`: String? = "ArraySubscript",
    var arrayValue: String?,
    var index: String?
) : Statement, Arguments, NameInterface {
    override fun getWrittenName(): String {
        return arrayValue!!
    }
}

File: Interpreter/Interpreter/Models/serialize/statementTypes/ArraySubscript.cs
namespace Interpreter.Models.serialize.statementTypes{
    public class ArraySubscript: IStatement, INameInterface
    {
        public string arrayValue { get; set; }
    
        public string index { get; set; }
        public string GetWrittenName()
        {
            return arrayValue;
        }
    }
};